pkg/controller/logstash/driver.go

```diff
@@ -37,6 +40,8 @@ type Params struct {
 
 	OperatorParams    operator.Parameters
 	KeystoreResources *keystore.Resources
+	UseTLS            bool               // Service of API Server uses TLS
+	APIServerConfig   *configs.APIServer // resolved API server config
```

Comment:
I was wondering whether we really need both `APIServerConfig` and `UseTLS`  especially given that we have a `SSLEnabled` field in `configs.APIServer`. I think it absolutely correct to validate the service TLS config from the CRD and the Logstash config don't contradict each other, as you do. But once that is done could we not just rely on `APIServerConfig`?